main.go

```diff
@@ -311,3 +321,23 @@ func run(ctrlConfig spirev1alpha1.ControllerManagerConfig, options ctrl.Options)
 
 	return nil
 }
+
+func autoDetectClusterDomain() string {
+	cname, err := net.LookupCNAME(k8sDefaultService)
+	if err != nil {
+		setupLog.Error(err, "unable to autodetect cluster domain")
+		return ""
+	}
+
```

Comment:
The regex approach is fine, but I wonder if this would be easier for folks to reason about:

```
domain := strings.TrimPrefix(cname, k8sDefaultService + ".")
if domain == cname {
    // handle error case since we did not have the expected prefix
}
// trim off optional trailing dot
domain = strings.TrimSuffix(trimmed, ".")
if domain == "" {
    // handle error case since there wasn't anything left after trimming
}
return domain
```

Reply:
Went with this suggestion, its easier to read than the regex definitely: https://github.com/spiffe/spire-controller-manager/pull/90/commits/b51634319ff9e5933c300fde85f573cdaa6e2f20